src/bp/nlu-core/training-pipeline.ts

```diff
@@ -371,64 +302,14 @@ const TrainSlotTagger = async (input: TrainStep, tools: Tools, progress: progres
   debugTraining('Training slot tagger')
   const slotTagger = new SlotTagger(tools.mlToolkit)
 
-  await slotTagger.train(
-    input.intents.filter(i => i.name !== NONE_INTENT),
-    input.list_entities
-  )
+  await slotTagger.train(input.intents, input.list_entities)
 
   debugTraining('Done training slot tagger')
   progress()
 
   return slotTagger.serialized
 }
 
-const TrainOutOfScope = async (input: TrainStep, tools: Tools, progress: progressCB): Promise<_.Dictionary<string>> => {
-  debugTraining('Training out of scope classifier')
-  const trainingOptions: sdk.MLToolkit.SVM.SVMOptions = {
-    c: [10], // so there's no grid search
-    kernel: 'LINEAR',
-    classifier: 'C_SVC',
-    seed: input.nluSeed
-  }
-
-  const noneUtts = _.chain(input.intents)
-    .filter(i => i.name === NONE_INTENT)
-    .flatMap(i => i.utterances)
-    .value()
-
-  if (!isPOSAvailable(input.languageCode) || noneUtts.length === 0) {
-    progress()
-    return {}
-  }
-
-  const oos_points = featurizeOOSUtterances(noneUtts, Object.keys(input.vocabVectors), tools)
-  let combinedProgress = 0
-
-  type ContextModel = [string, string]
-  const ctxModels: ContextModel[] = await Promise.map(input.ctxToTrain, async ctx => {
-    const in_ctx_scope_points = _.chain(input.intents)
-      .filter(i => i.name !== NONE_INTENT && i.contexts.includes(ctx))
-      .flatMap(i => featurizeInScopeUtterances(i.utterances, i.name))
-      .value()
-
-    const svm = new tools.mlToolkit.SVM.Trainer()
-    const model = await svm.train([...in_ctx_scope_points, ...oos_points], trainingOptions, p => {
-      combinedProgress += p / input.ctxToTrain.length
-      progress(combinedProgress)
-    })
-
-    return [ctx, model] as [string, string]
-  })
-
-  debugTraining('Done training out of scope')
-  progress(1)
-  return ctxModels.reduce((acc, cur) => {
-    const [ctx, model] = cur!
-    acc[ctx] = model
-    return acc
-  }, {} as _.Dictionary<string>)
-}
-
 const NB_STEPS = 6 // change this if the training pipeline changes
```

Comment:
I've broken the progress logic right here but I have already repaired it in another branch locally.